### DISCUSSION TITLE
serenity template 6.0.3, sergen latest 6.1.3 does not show item on grid list

### DISCUSSION THREAD

**Discussion by minhhungit** [serenity-is/Serenity#6536] [Help] [answered]
Hi team, @volkanceylan @VictorTomaili, as title, can you guys please check this

step to produce:
- install latest serenity template
- upgrade lastet sergen dotnet tool
- update nuget package

build success, no error on browser console

![image](https://user-images.githubusercontent.com/2279508/190878689-32b1218f-0c35-4e95-8779-85cadf8b65b4.png)

![image](https://user-images.githubusercontent.com/2279508/190878783-4570231b-ae26-4ac1-9b9f-6d173da53227.png)

List service return data but they were not shown on grid

All serenity nuget packages were upgraded

![image](https://user-images.githubusercontent.com/2279508/190878723-e0af5c5e-f92e-442d-87ec-38238ed45153.png)

```xml
<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup>
    <TargetFramework>net6.0</TargetFramework>
    <PreserveCompilationContext>true</PreserveCompilationContext>
    <AssemblyName>Serene1.Web</AssemblyName>
    <TypeScriptCompileBlocked>true</TypeScriptCompileBlocked>
    <SerenitySrc>..\..\Serenity\src\</SerenitySrc>
    <DotNetSergen>dotnet sergen</DotNetSergen>
    <DotNetSergenDir>$(SerenitySrc)Serenity.Net.CodeGenerator\bin"</DotNetSergenDir>
    <DotNetSergen Condition="Exists('$(DotNetSergenDir)sergen')">$(DotNetSergenDir)sergen</DotNetSergen>
    <DotNetSergen Condition="Exists('$(DotNetSergenDir)sergen.exe')">$(DotNetSergenDir)sergen.exe</DotNetSergen>
    <CommonSrc>..\..\common-features\src\</CommonSrc>
    <ProSrc>..\..\pro-features\src\</ProSrc>
    <TypeScriptToolsVersion>Latest</TypeScriptToolsVersion>
  </PropertyGroup>
  <ItemGroup>
    <EmbeddedResource Include="Migrations\**\*.sql" Exclude="bin\**;obj\**;**\*.xproj;packages\**;@(EmbeddedResource)" />
    <None Update="wwwroot\**\*">
      <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
    </None>
    <None Update="Modules\**\*.Template.html;Modules\**\*.ts.html">
      <CopyToPublishDirectory>PreserveNewest</CopyToPublishDirectory>
    </None>
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="FastMember" Version="1.5.0" />
    <PackageReference Include="FluentMigrator.Runner" Version="3.3.2" />
    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="6.0.2" />
    <PackageReference Include="Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation" Version="6.0.2" />
    <PackageReference Include="Microsoft.Data.Sqlite" Version="6.0.3" />
    <PackageReference Include="Microsoft.TypeScript.MSBuild" Version="4.6.2">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
    </PackageReference>
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Serenity.Assets" Version="6.1.3" />
    <PackageReference Include="Serenity.Scripts" Version="6.1.3" />
    <PackageReference Include="Serenity.Net.Web" Version="6.1.3" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Serenity.Extensions" Version="6.1.3" />
    <PackageReference Include="Serenity.Demo.Northwind" Version="6.1.3" />
    <PackageReference Include="Serenity.Demo.BasicSamples" Version="6.1.3" />
  </ItemGroup>
  <Import Project="$(SolutionDir)Serenity\build\submodule.targets" Condition="Exists('$(SolutionDir)Serenity\build\submodule.targets')" />
  <Target Name="CreateInputsOutputs" BeforeTargets="BeforeBuild">
    <CreateItem Include="Modules\**\*.ts;Imports\**\*.ts;tsconfig.json">
      <Output TaskParameter="Include" ItemName="CompileTSCInputs" />
    </CreateItem>
  </Target>
  <Target Name="NpmInstall" BeforeTargets="BeforeBuild" Condition="!Exists('node_modules\@types\jquery\index.d.ts')">
    <Message Text="Installing NPM packages..." Importance="high"></Message>
    <Exec Command="npm install" />
  </Target>
  <Target Name="TransformMvcClientTypes" BeforeTargets="BeforeBuild">
    <Exec Command="dotnet tool restore" ContinueOnError="true" />
    <Exec Command="$(DotNetSergen) restore" ContinueOnError="true" />
    <Exec Command="$(DotNetSergen) mvct" ContinueOnError="true" />
  </Target>
  <Target Name="TransformServerTypings" AfterTargets="AfterBuild">
    <Exec Command="$(DotNetSergen) servertypings" ContinueOnError="true" />
  </Target>
  <Target Name="CompileTSC" AfterTargets="AfterBuild" Inputs="@(CompileTSCInputs)" Outputs="wwwroot\Scripts\site\Serene1.Web.js">
    <Message Text="Executing TypeScript compiler..." Importance="high"></Message>
    <Exec Command="node &quot;$(TSJavaScriptFile.Replace('build\\..\tools\', 'tools\'))&quot; -p ./tsconfig.json" ContinueOnError="true" />
  </Target>
</Project>

```

**package.json**

```json
{
  "name": "serene.aspnetcore",
  "version": "1.0.0",
  "description": "Serene1 ASP.NET Core",
  "main": "index.js",
  "repository": {
    "type": "git",
    "url": "https://github.com/serenity-is/serene"
  },
  "dependencies": {
    "@types/jquery": "^2.0.39",
    "@types/jquery.blockui": "0.0.27",
    "@types/jquery.cookie": "^1.4.28",
    "@types/jquery.validation": "^1.13.32",
    "@types/jqueryui": "^1.11.32",
    "@types/jspdf": "^1.1.31",
    "@types/react": "^16.3.14",
    "@types/sortablejs": "^1.3.31",
    "@types/toastr": "^2.1.33",
    "vue": "^2.5.2"
  },
  "devDependencies": {
  },
  "scripts": {
    "less": "lessc ./wwwroot/Content/site/site.less ./wwwroot/content/site/site.css"
  },
  "author": "",
  "license": "ISC"
}
```

**Comment by DanielRuanova** [comment]
I have the same problem

**Comment by VictorTomaili** [comment]
Use previous sergen for now. 

`dotnet tool install sergen --version 6.0.8`

**Reply by VictorTomaili** [reply]
edit: dont need to use global variable

**Reply by VictorTomaili** [reply]
The latest version of the sergen can be found on this link
https://www.nuget.org/packages/sergen

**Comment by DanielRuanova** [comment]
I'm using Visual Studio 2022, with the latest Serene, Serenity and common-features source code.
Updated nuget packages.

But this solution has not worked for me or I have not been able to apply it

**Comment by VictorTomaili** [comment, marked as answer]
Use the latest version of sergen version 6.1.6

`dotnet tool update sergen`

**Comment by dougdsank** [comment]
I have the same issues. All my grids are empty after using sergen versions 6.1.6
![2022-09-21_20-14-27](https://user-images.githubusercontent.com/3937709/191631814-2ebc29ce-2226-429f-879b-02e30552e736.jpg)

**Comment by minhhungit** [comment]
@VictorTomaili  I updated to 6.1.6 both sergen tool and nuget packages, but not work for me, this time I got 2 errors

![image](https://user-images.githubusercontent.com/2279508/191716837-9daaceff-2ddd-44e0-8dfc-7180acff380f.png)

![image](https://user-images.githubusercontent.com/2279508/191717181-ca21e54f-d6ca-43ee-babd-747e3e393a40.png)

**Reply by VictorTomaili** [reply]
Your issue is not related to sergen. Check the error description.

**Reply by VictorTomaili** [reply]
Don't try to use the latest 6.1.6 serenity packages. Use just only the sergen. Or update your es version.

**Reply by minhhungit** [reply]
> Don't try to use the latest 6.1.6 serenity packages. Use just only the sergen. Or update your es version.

yes this works, thank you

**Comment by DanielRuanova** [comment]
Hi:
Today I have created a new project using the SERENE template and... finally, data appears in the grid.

I still don't know what the difference is, I guess having updated SERGEN before creating this new project.

![image](https://user-images.githubusercontent.com/46539253/191751901-2677aa6a-f176-49dd-989b-84328a073f5b.png)
-----
![image](https://user-images.githubusercontent.com/46539253/191752627-60bbda68-dd69-49a6-bed8-26a54029b846.png)

**Reply by vinnaykulkarni** [reply]
Yes new project works fine, however when we update the serenity packages the data disappears, and even reverting them back to the older version also didn't worked for me

**Reply by choss21** [reply]
I have the same problem, Not working update sergen :(

**Comment by choss21** [comment]
I have the same problem, Not working update sergen :(

**Comment by VictorTomaili** [comment]
If you experience this error
![image](https://user-images.githubusercontent.com/2023440/193813342-2cc5aace-2c30-4270-b00e-185c6c3de85f.png)

Update the target to ES6 in your tsconfig.json file.

```diff
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
-   "target": "ES5",
+   "target": "ES6",
    "typeRoots": [
```

**Comment by fatihtok** [comment]
i have the same problem, it doesn't work after update

**Comment by VictorTomaili** [comment]
Update to latest serenity scripts nuget package.

**Comment by choss21** [comment]
Thanks,  it works for me :D
